### ISSUE TITLE
docker login

### ISSUE THREAD

**Issue by idelvall** [earthly/earthly#1722]
Hi, is there a way of authenticating to a remote docker registry that does not require installing docker in the host?

**Comment by idelvall** [comment]
Forget it, docker is a prerequisite (🥇 for me)

**Comment by idelvall** [comment]
Ok, rephrasing my words:
"Is there any way to authenticate to ECR without having to install the AWS cli in the host"?
I would prefer to install AWS CLI within earthly, so this is also portable. And this is doable, I can install it in a task and get the ECR temporary token (docker user and password). The problem I am facing, is that it seems that now I am forced to move this information to the host, to run docker login, and then invoking a different earthly command for pushing, when allowing to call docker login from within earthly would make everything simpler.
Any help appreciated guys

**Comment by vladaionescu** [comment]
Using the credential helper is the best way right now: https://docs.earthly.dev/docs/guides/configuring-registries/aws-ecr#install-and-configure-the-ecr-credential-helper. There's currently no way to use credentials passed from inside of Earthly for access to registries, unfortunately (your use case requires that). But we plan to support it in the future.

**Comment by idelvall** [comment]
thanks Vlad, is there a way to access the host docker unix socket from earthly?

**Comment by vladaionescu** [comment]
No, not really. You can still interact with the host Docker via `LOCALLY` + `WITH DOCKER` though.

BTW you should join our Slack ;-) https://earthly.dev/slack

**Comment by rrjjvv** [comment]
> There's currently no way to use credentials passed from inside of Earthly for access to registries, unfortunately (your use case requires that). But we plan to support it in the future.

@vladaionescu Is the support you mention specific to registry auth?  Or are you speaking to a more general concept, say, being able to consume secrets that are created at runtime (not just earthly invocation time)?  I have a scenario where that would be convenient (but far from required).

**Comment by vladaionescu** [comment]
We plan to support both actually - for the latter, check out this proposal https://github.com/earthly/earthly/issues/1298

**Comment by vladaionescu** [comment]
Somewhat related issue: https://github.com/earthly/earthly/issues/1797

**Comment by Konubinix** [comment]
Can I ask about the status of this feature? It would definitely be a good step in reproducibily. 

In our CI, we have dynamic registries. We run some earthly targets to compute the registry url, then we get out of earthly to login to that registry and run again earthly to run our +ci target. It would be great to simply run the +ci target and it would be able to call the appropriate target to log into the registry.
